src/Data/Map/Internal.purs

```diff
@@ -155,6 +156,14 @@ instance functorWithIndexMap :: FunctorWithIndex k (Map k) where
       Node h s k v l r ->
         Node h s k (f k v) (go l) (go r)
 
+instance bifunctorMap :: Bifunctor Map where
+  bimap f g = go
+    where
+    go = case _ of
+      Leaf -> Leaf
+      Node h s k v l r ->
+        Node h s (f k) (g v) (go l) (go r)
```

Comment:
Unfortunately, this is not a sound implementation of `bimap`. It's not possible to naively map over keys because the ordering invariant wouldn't be preserved (basically the new Map will be completely broken). This is why `Set k` (which is really `Map k Unit`) can't have a Functor instance. Functor (and by extension Bifunctor) does not admit the necessary `Ord` constraint. So instead, `Set` exports a monormorphic `map` implementation. One would have to do the same for a `bimap` implementation on `Map`.

https://github.com/purescript/purescript-ordered-collections/blob/313ec0c3556854cd4e4752a77a86b102fc41ad05/src/Data/Set.purs#L99-L105

Reply:
Thank you for you quick reply! I see the reason why I cant define Bifunctor instance. You can close this PR.
BTW, thank you for your performance improvement on this package.